packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -857,6 +860,75 @@ export class GarbageCollector implements IGarbageCollector {
 		return this.summaryStateTracker.refreshLatestSummary(result);
 	}
 
+	/**
+	 * Process a GC message.
+	 * @param message - The GC message from the container runtime.
+	 * @param local - Whether it was send by this client.
+	 */
+	public processMessage(message: ContainerRuntimeGCMessage, local: boolean) {
+		switch (message.contents.type) {
+			case GarbageCollectionMessageType.Sweep: {
+				// Delete the nodes whose ids are present in the contents.
+				const sweepMessageContents = message.contents.contents;
+				this.deleteSweepReadyNodes(sweepMessageContents.deletedNodeIds);
+				break;
+			}
+			default: {
+				const compatBehavior = message.compatDetails?.behavior;
+				if (compatBehavior !== "Ignore") {
```

Comment:
If you take my suggestions on adjusting how the types are defined, you can put this compatBehavior check in a function that takes in `message.contents.type` as a param of type `never` which will also ensure all known cases are covered in this switch statement.